src/charm.py

```diff
@@ -539,7 +554,11 @@ def _configure(self, _):
 
         self.remote_write_provider.update_endpoint()
         self.grafana_source_provider.update_source(self.external_url)
-        self.unit.status = ActiveStatus()
+
+        if self._is_tls_enabled() and not self.container.exists(CERT_PATH):
```

Comment:
I think it would be more clear if we abstracted this to a `self._is_tls_ready()` method. 
Enabled but not ready -> push cert to container, then restart. Regardless of how we got there, this should be the reasonable course of action. Or what am I missing?

Reply:
I'm thinking of writing the cert only on `cert-changed` event, and:

|                                  | Relation present | Cert file on disk                  | Hypothetical `is_tls_ready() -> bool` |
|----------------------------------|------------------|------------------------------------|-----------------------------|
| Isolated charm                   | No               | No                                 | No                          |
| tls-cert rel just joined         | Yes              | No                                 | No                          |
| on-cert-changed (after rel join) | Yes              | Yes                                | Yes                         |
| cert revoked/expired             | Yes              | Yes (keep on disk even if expired) | Yes                         |
| relation departed / broken       | Yes              | ?                                  | No (how? code ordering...)  |
| any event after rel-broken       | No              | No                                  | No |

Wdyt?

Reply:
seems reasonable. I think departed/broken should immediately remove the cert from disk and from that moment onwards is_tls_ready should return false (assuming it checks the file presence synchronously). 
Or we refactor to use manifests, but I'm not convinced that will make code ordering less of a headache